src/include/miopen/graphapi/util.hpp

```diff
@@ -0,0 +1,266 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/graphapi/tensor.hpp>
+#include <miopen/graphapi/opgraph.hpp>
+
+#include <functional>
+#include <numeric>
+#include <string_view>
+#include <unordered_map>
+
+namespace miopen {
+namespace graphapi {
+
+inline std::string tensorIdAsStr(const Tensor* tens_ptr)
+{
+
+    int64_t id = tens_ptr->getId();
+    char* b    = reinterpret_cast<char*>(&id);
```

Comment:
It seems it doesn't matter, since `uint64_t` is just a container for 8 chars. In a typical flow it is filled from `std::string_view` and here we just construct that string back. But in a common case it can be broken even for the little endian. `c_str()` and c-style string operations will be broken with zero less significant byte and both C strings and C++ `streams` will suffer from special characters.

I think it should be either bulletproof or should not be a part of the library (not in `include/miopen`).
We can use it as it is in the test, but if it's a part of the library and can be misused - it will be misused.

Reply:
@CAHEK7 +1